src/bson.c

```diff
@@ -1139,7 +1145,7 @@ PHP_FUNCTION(fromJSON)
 		RETVAL_STRINGL((const char *) bson_get_data(&b), b.len, 1);
 		bson_destroy(&b);
 	} else {
-		RETURN_NULL();
+		phongo_throw_exception(PHONGO_ERROR_UNEXPECTED_VALUE TSRMLS_CC, "%s", error.message);
```

Comment:
Reasonably sure based on `bson_json_reader_read()`; however, it won't hurt to add a fall-back.